metricflow/test/snapshots/test_dataflow_to_sql_plan.py/SqlQueryPlan/DuckDB/test_simple_fill_0_with_categorical_dimension__plan0_optimized.sql

```diff
@@ -0,0 +1,22 @@
+-- Compute Metrics via Expressions
+SELECT
+  booking__is_instant
+  , COALESCE(bookings, 0) AS bookings_fill_0
```

Comment:
This is something we haven't talked about at all I don't think, but I was thinking about it while reading the code. I believe this is the correct approach, but we need to document it carefully.

This does not fill values for missing dimension values. If, for some reason, there were 0 bookings with is_instant = true we would render one row - `false, <number>`.

In cases where users are doing some kind of full scope group by, or using those rows as inputs into a derived metric, they would want the full value set represented. However, in cases where they were filtering out those values, they would not.

Therefore, we cannot fill 0s for all dimension combinations, we can only ensure that any case where an actual NULL shows up (rather than a total absence of values) for a measure the NULL is coalesced to the specified fill value. This seems appropriate given the name of the property, but it's good for us to be clear about it in docs and so forth.

Reply:
I think we did discuss this at standup yesterday! i.e., the fact that we obviously don't have a "dimension values spine" table to join to so we won't be filling those missing rows. So product is aware of this limitation (at least Nick is). But definitely agree that we need to document this clearly!!